### DISCUSSION TITLE
Custom easing, filters, auto loop feature, and keyframing warp transformation

### DISCUSSION THREAD

**Discussion by Patchwxrk** [keiwando/toonsquid#453] [Ideas]
custom easing - such as the graphs in alight motion with overshoot and  copying/pasting graphs

filters - such as in (gags) procreate dreams. preferably able to keyframe these too 

auto loop - like in roughanimator where you can select a part of a layer and loop it and edit the original and all the loop parts edits with it

keyframing warp transformation - haven't seen any apps do this yet, but i'd love to be able to tween using the warp feature so i can make more dynamic movements.

**Comment by Kosdaos1** [comment]
The loop function in Roughanimator is the most convenient loop I've seen in animation programs. In Toonsquid, this function is performed by the Symbols function, but this is not always convenient.

**Reply by jsllllc** [reply]
I agree. At risk of feature bloat (as you said, it can be accomplished with additional steps via symbols), a RA-style in-timeline loop setting would be very useful. Particularly in being able to adjust loop timing to match other layers, as you can't see other layers when editing timing within a symbol clip. My current way of getting around this (for complex scenes) requires exporting the symbols / layers separately and fixing the timing in premiere, which adds considerable time to the process.

**Comment by Cr8iveMike** [comment]
I just want to point it out since you talked about keyframing warp, that Procreate Dreams can do that already. I would love to see that in toonsquid if there were any plans.
